packages/preview-middleware-client/src/utils/version.ts

```diff
@@ -23,12 +24,36 @@ const minVersionInfo = {
 } as Readonly<Ui5VersionInfo>;
 
 /**
- * Retrieve the UI5 version from sap.ui.core library
+ * Check if the given version info is valid.
+ * @param versionInfo to check
+ * @throws Error if the version info is invalid
+ */
+function checkVersionInfo(versionInfo: Ui5VersionInfo): void {
+    if(isNaN(versionInfo.major) ||
+       isNaN(versionInfo.minor) ||
+       isNaN(versionInfo.patch ?? 0)) {
+        throw new Error('Invalid version info');
+    }
+}
+
+/**
+ * Retrieve the UI5 version.
+ * If no library is given, it will try to get the version from 'sap.ui.server.abap' or 'sap.ui.core'.
  *
+ * @param lib - (optional) specific library name to get the version from, e.g. 'sap.m'
  * @returns Ui5VersionInfo
  */
-export async function getUi5Version(): Promise<Ui5VersionInfo> {
-    let version = ((await VersionInfo.load({ library: 'sap.ui.core' })) as SingleVersionInfo)?.version;
+export async function getUi5Version(lib?: string): Promise<Ui5VersionInfo> {
+    let version: string | undefined;
+    //fallback 1: sap.ui.server.abap because its always released together with SAPUI5 and thus has the same version
+    //fallback 2: sap.ui.core because its always present in a SAPUI5 application
+    const libraries = [lib, 'sap.ui.server.abap', 'sap.ui.core'].filter(Boolean);
```

Comment:
I'm not sure if we need to complicate this logic that much, especially when it still can't guarantee that it will always be correct and that this is needed for only a very small subset of the cases (it should be only relevant for patch versions). 

I think we just have to accept the fact that there could be different library patch versions and do the appropriate checks - look which library version is needed for the particular case and check that. Using `sap.ui.core` as default library for version check still makes sense because majority of the cases thats exactly what we're interested.

Reply:
ok agree. Will adjust to lib parameter or sap.ui.core